customer-managed/azure/privatelink/README.md

```diff
@@ -0,0 +1,244 @@
+
+    ```
+    CLUSTER_POST_BODY=`cat << EOF
+    {
+    "cloud_provider": "CLOUD_PROVIDER_AZURE",
+    "connection_type": "CONNECTION_TYPE_PRIVATE",
+    "name": "azure-byoc-pl",
+    "resource_group_id": "$RESOURCE_GROUP_ID",
+    "network_id": "$NETWORK_ID",
+    "region": "uksouth",
+    "throughput_tier": "tier-1-azure-beta",
+    "type": "TYPE_BYOC",
+    "zones": ["uksouth-az1",  "uksouth-az2", "uksouth-az3"],
+    "redpanda_version": "24.1",
+    "azure_private_link": { 
+        "allowed_subscriptions": ["$AZURE_PRIVATE_LINK_SUBSCRIPTION_ID"],
+        "enabled": true,
+        "connect_console": true
+    }
+    }
+    EOF`
+
+    RP_ID=`curl -s -X POST "${HEADERS[@]}" \
+    -d "$CLUSTER_POST_BODY" $API_ENDPOINT/v1beta2/clusters | jq -r '.operation.metadata.cluster_id'`
+
+    # NOTE: this should have output
+    echo $RP_ID
+    ```
+1. Run the rpk apply to create the infrastructure for the cluster. This will take about 45 minutes to complete. 
+    ```
+    rpk login --save --client-id=$CLOUD_CLIENT_ID --client-secret=$CLOUD_CLIENT_SECRET
+
+    rpk cloud byoc azure apply --redpanda-id=$RP_ID --subscription-id=$AZURE_SUBSCRIPTION_ID
+    
+    ```
+
+## Connecting to Redpanda Private Link
+
+Once the cluster has been created we can now run terraform to connect to the created cluster with private link. This terraform will provision a vm along with a private link endpoint that connects to the redpanda cluster. 
+
+
+
+1. First fetch credentials to the Redpanda cluster:
+
+    ```
+    az login 
+    az account set --subscription $AZURE_SUBSCRIPTION_ID
+    az aks get-credentials -g rg-rpcloud-$RP_ID -n aks-rpcloud-$RP_ID
+    secret=$(kubectl get secret redpanda-superusers -n redpanda -o jsonpath='{.data.users\.txt}' | base64 --decode)
+
+    IFS=':' read -r user password sasl <<< "$secret"
+
+    export USER=$user
+    export PASS=$password
+
+    ```
+1. Fetch outputs from the redpanda cluster that we will use as inputs to our terraform
+    ```
+    DNS_RECORD=`curl -s -X GET "${HEADERS[@]}" \
+    $API_ENDPOINT/v1beta2/clusters/$RP_ID | jq -r ".cluster.azure_private_link.status.dns_a_record"`
+
+    PRIVATE_SERVICE_ID=`curl -s -X GET "${HEADERS[@]}" \
+    $API_ENDPOINT/v1beta2/clusters/$RP_ID | jq -r ".cluster.azure_private_link.status.service_id"`
+
+    CONSOLE_URL=`curl -s -X GET "${HEADERS[@]}" \
+    $API_ENDPOINT/v1beta2/clusters/$RP_ID | jq -r ".cluster.redpanda_console.url"`
+
+    echo $DNS_RECORD
+    echo $PRIVATE_SERVICE_ID
+    echo $CONSOLE_URL
+
+    ```
+1. Next make sure you are logged into the subscription that was allowed in the creation request: 
+    ```
+    az login
+    az account set --subscription $AZURE_PRIVATE_LINK_SUBSCRIPTION_ID
+    ```
+1. Write our terraform variables out to a file so we can pick it up and use it
+    ```
+    export MYGROUP=your-rg-group
+    cat << EOF > terraform.tfvars
+    region = "uksouth"
```

Comment:
suggestion, nonblocking: maybe we should define $REGION and $AZS to parameterize the example